neuro_san/service/generic/async_agent_service.py

```diff
@@ -125,6 +125,12 @@ def get_request_timeout_seconds(self) -> float:
         """
         return self.agent_network_provider.get_agent_network().get_request_timeout_seconds()
 
+    def is_mcp_tool(self) -> bool:
+        """
+        :return: True if the agent served by this service is an MCP tool
+        """
+        return self.agent_network_provider.get_agent_network().is_mcp_tool()
```

Comment:
FWIW: "MCP tool" might not be the best long-term name for this. What you have is fine for now, though, to get going.
We're not really dealing with "tools" at this level of the service, we're merely trying to serve something up.
Maybe is_mcp_agent() instead or is_served_by_mcp()? or maybe just is_mcp()?